redisvl/query/hybrid.py

```diff
@@ -0,0 +1,363 @@
+from typing import Any, Dict, List, Literal, Optional, Set, Union
+
+from redis.commands.search.query import Filter
+
+from redisvl.redis.utils import array_to_buffer
+from redisvl.utils.full_text_query_helper import FullTextQueryHelper
+
+try:
+    from redis.commands.search.hybrid_query import (
+        CombinationMethods,
+        CombineResultsMethod,
+        HybridPostProcessingConfig,
+    )
+    from redis.commands.search.hybrid_query import HybridQuery as RedisHybridQuery
+    from redis.commands.search.hybrid_query import (
+        HybridSearchQuery,
+        HybridVsimQuery,
+        VectorSearchMethods,
+    )
+except ImportError:
+    raise ImportError(
+        "Hybrid queries require Redis Open Source >= 8.4.0 and redis-py>=7.1.0"
+    )
+
+from redisvl.query.filter import FilterExpression
+
+
+class HybridQuery:
+    """
+    A hybrid search query that combines text search and vector similarity, with configurable fusion methods.
+
+    .. code-block:: python
+
+        from redisvl.query.hybrid import HybridQuery
+        from redisvl.index import SearchIndex
+
+        index = SearchIndex.from_yaml("path/to/index.yaml")
+
+        query = HybridQuery(
+            text="example text",
+            text_field_name="text_field",
+            vector=[0.1, 0.2, 0.3],
+            vector_field_name="vector_field",
+            text_scorer="BM25STD",
+            yield_text_score_as="text_score",
+            yield_vsim_score_as="vector_similarity",
+            combination_method="LINEAR",
+            linear_alpha=0.3,
+            yield_combined_score_as="hybrid_score",
+            num_results=10,
+            return_fields=["field1", "field2"],
+            stopwords="english",
+        )
+
+        results = index.hybrid_search(query)
+
+    See Also:
+        - `FT.HYBRID command documentation <https://redis.io/docs/latest/commands/ft.hybrid>`_
+        - `redis-py hybrid_search documentation <https://redis.readthedocs.io/en/stable/redismodules.html#redis.commands.search.commands.SearchCommands.hybrid_search>`_
+    """
+
+    def __init__(
+        self,
+        text: str,
+        text_field_name: str,
+        vector: Union[bytes, List[float]],
+        vector_field_name: str,
+        text_scorer: str = "BM25STD",
+        text_filter_expression: Optional[Union[str, FilterExpression]] = None,
+        yield_text_score_as: Optional[str] = None,
+        vector_search_method: Optional[Literal["KNN", "RANGE"]] = None,
+        knn_k: Optional[int] = None,
+        knn_ef_runtime: Optional[int] = None,
+        range_radius: Optional[int] = None,
+        range_epsilon: Optional[float] = None,
+        yield_vsim_score_as: Optional[str] = None,
+        vector_filter_expression: Optional[Union[str, FilterExpression]] = None,
```

Comment:
```suggestion
        filter_expression: Optional[Union[str, FilterExpression]] = None,
```
for consistency with our other query objects